lib/hex_mirror/mirror.ex

```diff
@@ -33,7 +34,18 @@ This module is responsible for actually downloading the hex registry, and then d
           File.write!(version_path(package, version), body)
           IO.puts "Downloading #{package} #{version}"
           {:ok, :downloaded}
+        {:error, %HTTPoison.Error{id: id, reason: reason}} ->
+          IO.puts "*** Error #{id}: downloading #{package} #{version}"
+          IO.puts "*** Reason: #{reason}"
       end
     end
   end
+
+  defp ensure_tarball_dir do
+    path = tarball_path
+    case File.mkdir_p(path) do
+      :ok -> path
+      {:error, _} -> nil
```

Comment:
Whats the reason for trapping the error here and then throwing it away?

Wouldn't an explicit explosion be easier to debug?